niceman/resource/session.py

```diff
@@ -375,19 +521,57 @@ def read(self, path):
     def mkdir(self, path, parents=False):
         """Create a directory
         """
-        self.execute_command(["mkdir"] + (["-p"] if parents else [""]) + [path])
+        command = ["mkdir"]
+        if parents: command.append("-p")
+        command += [path]
+        self.execute_command(command)
 
         if not self.isdir(path):
             raise CommandError(cmd='mkdir', msg="Failed to create directory")
 
     def isdir(self, path):
         try:
-            self.execute_command(['test', '-d', path])
+            out, err = self.execute_command(['test', '-d', path])
+        except Exception as exc:  # TODO: More specific exception?
+            lgr.debug("Check for directory failed: %s", exc_str(exc))
+            return False
+        if not err:
             return True
-        except CommandError:
+        else:
+            lgr.debug("Standard error was not empty (%r), thus assuming that "
+                      "test for directory has failed", err)
             return False
-    # chmod?
-    # chown?
+
+    def chmod(self, path, mode, recursive=False):
+        """Set the mode of a remote path
+        """
+        command = ['chmod']
+        if recursive: command += ["-R"]
+        command += [mode] + [path]
+        self.execute_command(command)
+
+    def chown(self, path, uid=-1, gid=-1, recursive=False):
+        """Set the user and gid of a path
+        """
+        command = ['chown']
+        if recursive: command += ["-R"]
+        if int(uid) > 0 and int(gid) > 0: command += ["{}.{}".format(uid, gid)]
```

Comment:
root has id 0 and we should allow to change to it as well. So, `if uid > -1` as well I guess